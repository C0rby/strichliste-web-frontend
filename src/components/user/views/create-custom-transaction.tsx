import { Card } from 'bricks-of-sand';
import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { BackButton } from '../../common';
import { ConnectedCreateCustomTransactionForm } from '../../transaction/create-custom-transaction-form';
import { FixedFooter, Section } from '../../ui';

export function CreateCustomTransaction(
  props: RouteComponentProps<{ id: string; deposit: string }>
): JSX.Element {
  return (
    <>
      <Section>
        <Card>
          <ConnectedCreateCustomTransactionForm
            transactionCreated={() => props.history.goBack()}
            userId={Number(props.match.params.id)}
          />
        </Card>
      </Section>
      <FixedFooter>
        <BackButton />
      </FixedFooter>
    </>
  );
}
