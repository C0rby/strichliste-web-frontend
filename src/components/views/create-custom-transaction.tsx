import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { BackButton } from '../common';
import { ConnectedCreateCustomTransactionForm } from '../transaction/create-custom-transaction-form';
import { Card, FixedFooter, Section } from '../ui';

export function CreateCustomTransaction(
  props: RouteComponentProps<{ id: number; deposit: string }>
): JSX.Element {
  return (
    <>
      <Section>
        <Card>
          <ConnectedCreateCustomTransactionForm
            transactionCreated={() => props.history.goBack()}
            isDeposit={props.match.params.deposit === 'deposit'}
            userId={props.match.params.id}
          />
        </Card>
      </Section>
      <FixedFooter>
        <BackButton />
      </FixedFooter>
    </>
  );
}
