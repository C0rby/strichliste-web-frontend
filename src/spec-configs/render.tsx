import { ThemeProvider, theme } from 'bricks-of-sand';
import { MemoryHistory, createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { RenderResult, render } from 'react-testing-library';
import { DeepPartial, Store, createStore } from 'redux';
import { StoreContext } from 'redux-react-hook';

import { AppState, reducer } from '../store';

const themeConfig = { light: theme, dark: theme };

export function renderWithContext(
  ui: JSX.Element,
  initialState: DeepPartial<AppState>,
  store: Store<AppState> = createStore(reducer, initialState),
  _route: string = '/',
  history: MemoryHistory = createMemoryHistory()
): RenderResult {
  return render(
    <ThemeProvider themes={themeConfig}>
      <StoreContext.Provider value={store}>
        <Provider store={store}>
          <Router history={history}>
            <IntlProvider>{ui}</IntlProvider>
          </Router>
        </Provider>
      </StoreContext.Provider>
    </ThemeProvider>
  );
}

export function renderAndReturnContext(
  ui: JSX.Element,
  initialState: DeepPartial<AppState>,
  store: Store<AppState> = createStore(reducer, initialState),
  _route: string = '/',
  history: MemoryHistory = createMemoryHistory()
): { result: RenderResult; store: Store<AppState>; history: MemoryHistory } {
  return {
    result: render(
      <ThemeProvider themes={themeConfig}>
        <Provider store={store}>
          <Router history={history}>
            <IntlProvider textComponent={React.Fragment}>{ui}</IntlProvider>
          </Router>
        </Provider>
      </ThemeProvider>
    ),
    store,
    history,
  };
}

export function renderWithIntl(ui: JSX.Element): RenderResult {
  return render(
    <ThemeProvider themes={themeConfig}>
      <IntlProvider textComponent={React.Fragment}>{ui}</IntlProvider>
    </ThemeProvider>
  );
}
