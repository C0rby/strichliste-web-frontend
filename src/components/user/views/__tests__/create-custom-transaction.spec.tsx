// tslint:disable:no-any

import * as React from 'react';
import { cleanup } from 'react-testing-library';

import { renderWithContext } from '../../../../spec-configs/render';
import { CreateCustomTransaction } from '../create-custom-transaction';

afterEach(cleanup);

const mock: any = jest.fn();

describe('CreateCustomTransaction', () => {
  describe('with deposit', () => {
    it('matches the snapshot', () => {
      const { container } = renderWithContext(
        <CreateCustomTransaction
          history={mock}
          match={{ params: { id: 5, deposit: 'deposit' } } as any}
          location={mock}
        />,
        { user: { 5: { balance: 0 } } }
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with dispense', () => {
    it('matches the snapshot', () => {
      const { container } = renderWithContext(
        <CreateCustomTransaction
          history={mock}
          match={{ params: { id: 5, deposit: 'dispense' } } as any}
          location={mock}
        />,
        { user: { 5: { balance: 0 } } }
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
