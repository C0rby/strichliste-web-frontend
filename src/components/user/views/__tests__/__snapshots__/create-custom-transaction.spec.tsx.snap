// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCustomTransaction with deposit matches the snapshot 1`] = `
<section
  class="css-q5fqw0"
>
  <div
    class="css-kmcglx"
  >
    <div
      class="css-1aayd4f"
    >
      <button
        disabled=""
        type="submit"
        class="css-8yq2qw"
      >
        -
      </button>
      <input
        placeholder="CUSTOM AMOUNT"
        type="text"
        value="0.00"
      />
      <button
        disabled=""
        type="submit"
        class="css-8yq2qw"
      >
        +
      </button>
    </div>
  </div>
</section>
`;

exports[`CreateCustomTransaction with dispense matches the snapshot 1`] = `
<section
  class="css-q5fqw0"
>
  <div
    class="css-kmcglx"
  >
    <div
      class="css-1aayd4f"
    >
      <button
        disabled=""
        type="submit"
        class="css-8yq2qw"
      >
        -
      </button>
      <input
        placeholder="CUSTOM AMOUNT"
        type="text"
        value="0.00"
      />
      <button
        disabled=""
        type="submit"
        class="css-8yq2qw"
      >
        +
      </button>
    </div>
  </div>
</section>
`;
